docs/IR.md

```diff
@@ -86,9 +86,15 @@ Each model has the following components:
 |doc_string|string|A human-readable documentation for this model. Markdown is allowed.|
 |graph|Graph|The parameterized graph that is evaluated to execute the model.|
 |metadata_props|map<string,string>|Named metadata values; keys should be distinct.|
+|external_models|ExternalModel|A set of model identifiers `(domain,name,version)` for imported external models referenced within the current model.|
 
- Models MUST specify a domain and use reverse domain names based on the responsible organization's identity, the same convention that is traditionally used for naming Java packages.
+Models MUST specify a domain and use reverse domain names based on the responsible organization's identity, the same convention that is traditionally used for naming Java packages.
  
+#### Model Composition
+
+External, explicitly imported, models may be treated as custom operators within other models. For such external model references to be resolved, each external model dependency MUST be declared in the 'external_models' set. External models are uniquely identified by their domain, the name of the main graph, and the model version. Model recursion is not supported, so model references must form acyclic graphs.
```

Comment:
We should either (a) restrict the list so that there is at most one entry for each (domain,name) pair or (b) indicate which one will prevail if multiple references exist. For now (a) may be simpler.

Reply:
Fixed